File: QuantifiedStrides/models/checkin.py
from datetime import date
from typing import Literal

from pydantic import BaseModel, Field


# ---------------------------------------------------------------------------
# Daily readiness — POST body
# ---------------------------------------------------------------------------

class DailyReadinessCreateSchema(BaseModel):
    entry_date: date
    overall_feel: int = Field(ge=1, le=10)
    legs_feel: int = Field(ge=1, le=10)
    upper_body_feel: int = Field(ge=1, le=10)
    joint_feel: int = Field(ge=1, le=10)
    injury_note: str | None = None
    time_available: Literal["short", "medium", "long"]
    going_out_tonight: bool = False


class DailyReadinessSchema(BaseModel):
    readiness_id: int
    user_id: int
    entry_date: date
    overall_feel: int
    legs_feel: int
    upper_body_feel: int
    joint_feel: int
    injury_note: str | None
    time_available: Literal["short", "medium", "long"] | None
    going_out_tonight: bool


# ---------------------------------------------------------------------------
# Workout reflection — POST body
# ---------------------------------------------------------------------------

class WorkoutReflectionCreateSchema(BaseModel):
    entry_date: date
    session_rpe: int = Field(ge=1, le=10)
    session_quality: int = Field(ge=1, le=10)
    notes: str | None = None
    load_feel: int | None = Field(default=None, ge=-2, le=2)
    workout_id: int | None = None


class WorkoutReflectionSchema(BaseModel):
    reflection_id: int
    user_id: int
    entry_date: date
    session_rpe: int
    session_quality: int
    notes: str | None
    load_feel: int | None
    workout_id: int | None = None


# ---------------------------------------------------------------------------
# Journal entry — POST body
# ---------------------------------------------------------------------------

class JournalEntryCreateSchema(BaseModel):
    entry_date: date
    content: str = Field(min_length=1)


class JournalEntrySchema(BaseModel):
    entry_id: int
    user_id: int
    entry_date: date
    content: str


# ---------------------------------------------------------------------------
# Journal history row (joined: readiness + reflection + journal)
# ---------------------------------------------------------------------------

class JournalHistoryRowSchema(BaseModel):
    entry_date: date
    overall: int | None
    legs: int | None
    upper: int | None
    joints: int | None
    injury_note: str | None
    time_available: Literal["short", "medium", "long"] | None
    going_out: bool | None
    rpe: int | None
    session_quality: int | None
    load_feel: int | None
    reflection_notes: str | None
    journal_note: str | None

File: mobile/mobile/src/api/endpoints/checkin.ts
import { apiClient } from '../client'

// ── Readiness ──────────────────────────────────────────────────────────────────

export interface ReadinessPayload {
  entry_date:        string
  overall_feel:      number
  legs_feel:         number
  upper_body_feel:   number
  joint_feel:        number
  injury_note?:      string | null
  time_available:    'short' | 'medium' | 'long'
  going_out_tonight: boolean
}

export async function saveReadiness(payload: ReadinessPayload): Promise<void> {
  await apiClient.post('/checkin/readiness', payload)
}

export async function fetchReadiness(date: string): Promise<ReadinessPayload | null> {
  try {
    const res = await apiClient.get(`/checkin/readiness/${date}`)
    return res.data
  } catch {
    return null
  }
}

// ── Workout reflection ────────────────────────────────────────────────────────

export interface WorkoutReflectionPayload {
  entry_date:      string
  session_rpe:     number
  session_quality: number
  notes:           string | null
  load_feel:       number | null
  workout_id:      number | null
  session_id:      number | null
}

export async function saveWorkoutReflection(
  payload: WorkoutReflectionPayload,
): Promise<void> {
  await apiClient.post('/checkin/reflection', payload)
}
